src/pg_shard.c

```diff
@@ -2104,50 +2111,8 @@ PgShardProcessUtility(Node *parsetree, const char *queryString,
 	}
 	else if (statementType == T_CopyStmt)
 	{
-		CopyStmt *copyStatement = (CopyStmt *) parsetree;
-		RangeVar *relation = copyStatement->relation;
-		Node *rawQuery = copyObject(copyStatement->query);
-
-		if (relation != NULL)
-		{
-			bool failOK = true;
-			Oid tableId = RangeVarGetRelid(relation, NoLock, failOK);
-			bool isDistributedTable = false;
-
-			Assert(rawQuery == NULL);
-
-			isDistributedTable = IsDistributedTable(tableId);
-			if (isDistributedTable)
-			{
-				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
Frankly speaking I do not think that it is good idea: I prefer to have all logic related with handling  copy in one place. 
Moreover, splitting logic will cause some code duplication: for example extraction of tableId.   
Also, you asked me to put all new code in new file...